aws/resource_aws_client_vpn_endpoint.go

```diff
@@ -0,0 +1,279 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsClientVpnEndpoint() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsClientVpnEndpointCreate,
+		Read:   resourceAwsClientVpnEndpointRead,
+		Delete: resourceAwsClientVpnEndpointDelete,
+		Update: resourceAwsClientVpnEndpointUpdate,
+		Importer: &schema.ResourceImporter{
+			State: resourceAwsClientVpnEndpointImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"client_cidr_block": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"dns_servers": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"server_certificate_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"transport_protocol": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "udp",
```

Comment:
We should check the user has provided a valid value:

```go
				ValidateFunc: validation.StringInSlice([]string{
					ec2.TransportProtocolTcp,
					ec2.TransportProtocolUdp,
				}, false),
```

Reply:
Done (using the actual values instead of these example stucts)